include-build/roborazzi-gradle-plugin/src/main/java/io/github/takahirom/roborazzi/RoborazziPlugin.kt

```diff
@@ -261,27 +272,35 @@ abstract class RoborazziPlugin : Plugin<Project> {
       testTaskProvider
         .configureEach { test: AbstractTestTask ->
           val resultsDir = resultDirFileProperty.get().asFile
-          if (restoreOutputDirRoborazziTaskProvider.isPresent) {
-            // Previous outputs are an input when running in compare or verify mode.
-            // However, during record runs the output dir might not exist yet, so we use
-            // files() to express that it is optional.
-            // See also: https://github.com/gradle/gradle/issues/2016
-            test.inputs.files(restoreOutputDirRoborazziTaskProvider.map {
-              if (!it.outputDir.get().asFile.exists()) {
-                it.outputDir.get().asFile.mkdirs()
-              }
-              test.infoln("Roborazzi: Set input dir ${it.outputDir.get()} to test task")
-              it.outputDir.files(".")
-            })
-          } else {
-            test.inputs.files(outputDir.map {
-              if (!it.asFile.exists()) {
-                it.asFile.mkdirs()
+          test.inputs.files(
```

Comment:
I didn't know we can pass a `Provider` to `files`! (In fact, this is roughly what I was trying to do originally, but instead I was calling `isRecordRun.get()` which fails.) 👍